src/ai/backend/common/events/dispatcher.py

```diff
@@ -420,71 +469,45 @@ async def _consume_loop(self) -> None:
         async for msg in self._msg_queue.consume_queue():  # type: ignore
             if self._closed:
                 return
-            event_type = "unknown"
-            start = time.perf_counter()
+            decoded_event_name = msg.payload[b"name"].decode()
+
+            async def done() -> None:
+                # To ensure that all consumer handlers are called.
+                # Basically there should be only one consumer handler for one event.
+                handler_count = len(self._subscribers[decoded_event_name])
+                async with self._msg_id_consumed_count_lock:
+                    self._msg_id_consumed_count[msg.msg_id] += 1
+                    if self._msg_id_consumed_count[msg.msg_id] < handler_count:
```

Comment:
Since we're only doing synchronous logic, there's no need to lock it, right?

Reply:
the consumer loop creates async tasks so I think we need async lock